src/dbnode/storage/shard.go

```diff
@@ -1132,6 +1145,46 @@ func (s *dbShard) ReadEncoded(
 	return reader.ReadEncoded(ctx, start, end, nsCtx)
 }
 
+func (s *dbShard) IndexChecksum(
```

Comment:
I suggest `FetchIndexChecksum` (to better match my suggested name change for `StreamIndexChecksum`), here, and in other places along the stack.